workspace_tools/tests.py

```diff
@@ -557,6 +557,14 @@
         "automated": True,
         #"host_test": "wait_us_auto"
     },
+    {
+        "id": "MBED_35", "description": "SPI C12832 display",
+        "source_dir": join(TEST_DIR, "mbed", "spi_C12832"),
+        "dependencies": [MBED_LIBRARIES, TEST_MBED_LIB, join(PERIPHERALS, 'C12832')],
```

Comment:
About adding new libraries, you should at least:
* Define new library (```'C12832'```) metadata in ```libraries.py```.
* Use ```paths.py``` to define library constant & paths to your new library (sources and headers), note path to library and its constant (e.g. ```CPPUTEST_LIBRARY```) ends with ```_LIBRARY```. In your case it should be ```C12832_LIBRARY```:

```
"dependencies": [MBED_LIBRARIES, TEST_MBED_LIB, join(PERIPHERALS, C12832_LIBRARY)],
```

Note: There is another option, you can just make ```C12832``` drivers' code as part of this test (maybe not the best idea).